cmd/utils/jeth.go

```diff
@@ -56,44 +57,52 @@ func (self *Jeth) err(call otto.FunctionCall, code int, msg string, id *int64) (
 	return res
 }
 
-// UnlockAccount asks the user for the password and than executes the jeth.UnlockAccount callback in the jsre
+// UnlockAccount asks the user for the password and than executes the jeth.UnlockAccount callback in the jsre.
+// It will need the public address for the account to unlock as first argument.
+// The second argument is an optional string with the password. If not given the user is prompted for the password.
+// The third argument is an optional integer which specifies for how long the account will be unlocked (in seconds).
 func (self *Jeth) UnlockAccount(call otto.FunctionCall) (response otto.Value) {
-	var account, passwd string
-	timeout := int64(300)
-	var ok bool
+	var account, passwd otto.Value
+	duration := otto.NullValue()
 
-	if len(call.ArgumentList) == 0 {
-		fmt.Println("expected address of account to unlock")
+	if !call.Argument(0).IsString() {
+		fmt.Println("first argument must be the account to unlock")
 		return otto.FalseValue()
 	}
 
-	if len(call.ArgumentList) >= 1 {
-		if accountExport, err := call.Argument(0).Export(); err == nil {
-			if account, ok = accountExport.(string); ok {
-				if len(call.ArgumentList) == 1 {
-					fmt.Printf("Unlock account %s\n", account)
-					passwd, err = PromptPassword("Passphrase: ", true)
-					if err != nil {
-						return otto.FalseValue()
-					}
-				}
-			}
+	account = call.Argument(0)
+
+	// if password is not given or as null value -> ask user for password
+	if call.Argument(1).IsUndefined() || call.Argument(1).IsNull() {
+		fmt.Printf("Unlock account %s\n", account)
+		if password, err := PromptPassword("Passphrase: ", true); err == nil {
+			passwd, _ = otto.ToValue(password)
+		} else {
+			throwJSExeception(err.Error())
 		}
 	}
-	if len(call.ArgumentList) >= 2 {
-		if passwdExport, err := call.Argument(1).Export(); err == nil {
-			passwd, _ = passwdExport.(string)
+
+	// verify that if a second non null argument was given it was a string
+	if call.Argument(1).IsDefined() && !call.Argument(1).IsNull() {
```

Comment:
This could simple be `else`. I think it's more readable than having to understand the conditions again. Also probably more future proof rather than having to maintain te exact negation of the previous if condition.